File: lib/alertSounds.ts
// Selectable interval-change alert sounds. Static require()s so Metro bundles them.
export interface AlertSound {
  id: string;
  label: string;
  src: number;
}

export const ALERT_SOUNDS: AlertSound[] = [
  { id: "beep", label: "Beep", src: require("../assets/sounds/alert-beep.wav") },
  { id: "chime", label: "Chime", src: require("../assets/sounds/alert-chime.wav") },
  { id: "bell", label: "Bell", src: require("../assets/sounds/alert-bell.wav") },
  { id: "ding", label: "Ding", src: require("../assets/sounds/alert-ding.wav") },
  { id: "pulse", label: "Pulse", src: require("../assets/sounds/alert-pulse.wav") },
];

export const DEFAULT_ALERT_ID = "beep";

File: IntervalTimer/Audio/AlertSounds.swift
import Foundation

struct AlertSound: Identifiable {
    let id: String
    let label: String
    /// Bundled resource filename (without extension); all are .wav.
    let file: String
}

enum AlertSounds {
    static let all: [AlertSound] = [
        AlertSound(id: "beep", label: "Beep", file: "alert-beep"),
        AlertSound(id: "chime", label: "Chime", file: "alert-chime"),
        AlertSound(id: "bell", label: "Bell", file: "alert-bell"),
        AlertSound(id: "ding", label: "Ding", file: "alert-ding"),
        AlertSound(id: "pulse", label: "Pulse", file: "alert-pulse"),
    ]

    static let defaultId = "beep"

    static func file(for id: String) -> String {
        all.first { $0.id == id }?.file ?? "alert-beep"
    }
}
